### ISSUE TITLE
Remove buildBasePath support for MSBuild projects

### ISSUE THREAD

**Issue by NTaylorMullen** [dotnet/cli#4534]
In old project.json land buildBasePath used to be a first class concept that can be modified by a single parameter. MSBuild no longer has this capability (requires multiple changes to get the same effect).

Due to this change in expectations we should nuke the CLI.Utils API points that allow a user to specify `buildBasePath` (since it can't be easily specified in CSProj).

**Comment by nguerrera** [comment]
Making another SDK change, and it's just begging to add a BaseOutputPath (default bin\) for symmetry with BaseIntermediateOutputPath (default obj\).

I think it is useful to support setting only BaseOutputPath and letting the SDK decide the defaults under it and we should do it. The CLI can set BaseOutputPath based on buildBasePath once that's done.

**Comment by NTaylorMullen** [comment]
So build base path will be supported soon?

**Comment by nguerrera** [comment]
I don't know about "soon" in the CLI. I will have a building block in the SDK shortly.

What were the precise semantics of buildBasePath in CLI? 

Did it:
* also redirect intermediate output?
* still make each project build to its own folder?

If so, matching the semantics would not be as trivial as setting /p:BaseOutputPath={buildBasePath}, It would have to be something like /p:BaseOutputPath={buildBasePath}\{projectName} and ditto for IntermediateOutputPath...

**Comment by NTaylorMullen** [comment]
> also redirect intermediate output?

I believe so but would need to verify.



> still make each project build to its own folder?

Build base path would be set to something like `../../somewhere` and then you'd have a `../../somewhere/ProjectName/...` structure. The project built with a build base path would not output any bits in the folder with the project file

**Comment by livarcocc** [comment]
We have issue https://github.com/dotnet/cli/issues/5020 which I believe is very close to this one.

The real problem here is not just allowing for buildBasePath, but to allow for arbitrary MSBuild properties to be passed when evaluating the project to find the tool.

**Comment by blackdwarf** [comment]
Is this fixed in the latest? We have the ability to pass in arbitrary MSBuild properties now using `/p:<param>:<value>`.

**Comment by livarcocc** [comment]
@blackdwarf No, this needs us to change the APIs so that it supports passing arbitrary msbuild parameters to the tool.

We need to figure out the format that we are going to use for that when invoking the tool.

We are not going to get to this for RTM and I think the change at this point would be someone big.
